feat: LEAP-1924: Allow to edit Timeline span by drag-n-drop

hlomzik opened the pull request:
Allows smooth intuitive interaction to easily edit spans on timeline.

The biggest effort here was to fix undo/redo stack to keep UX convenient and predictable. Only one item in history is created with final state for these actions:
- new region is drawn
- existing region was edited by drag-n-drop
- existing region was edited in info panel:
  - in input and the focus was moved away
  - in input by clicking on up/down arrows

`Video#startDrawing()`, `Video#finishDrawing()` and `TimelineRegion#setRange()` (former less correct `setRanges()`) were edited to accept `mode` to distinguish actions needed and `region` to find current region to edit it instead of creating a new one.

The main entry point is still `Timeline/Views/Frames#onFrameScrub()` — now extended to handle drawing new region, editing existing region and updating current position in a Video tag.

hlomzik committed:
feat: LEAP-1924: Allow to edit Timeline span by drag-n-drop

Smooth intuitive interaction to easily edit spans on timeline.

The biggest effort here was to fix undo/redo stack to keep UX convenient and predictable.
Only one item in history is created with final state for these actions:
- new region is drawn
- existing region was edited by drag-n-drop
- existing region was edited in info panel:
  - in input and the focus was moved away
  - in input by clicking on up/down arrows

hlomzik committed:
Add more comments about params in Video methods

hlomzik committed:
Update values in info panel when user edits the region

hlomzik committed:
Fix param name to be more consistent with its type

Gondragos commented on web/libs/editor/src/components/Timeline/Views/Frames/Frames.tsx:
```diff
@@ -232,15 +256,19 @@ export const Frames: FC<TimelineViewProps> = ({
         if (region) {
-          const [start, end] = frame > baseFrame ? [baseFrame, frame] : [frame, baseFrame];
-          region.setRanges([start, end]);
+          if (isInstant) {
```
Potentially missed setting `isInstant` to `true`. At least I can't find it for now.

hlomzik committed:
Fix accidentially removed instant regions handling

hlomzik committed:
Show resize cursor on hover

Too many changes for such a small thing, but it improves UX

hlomzik committed:
Fix linting

Gondragos commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13964875656)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-leap-1924/edit-timeline-spans'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13964875656

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13964875656

hlomzik committed:
Fix linting (again)

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13986565837)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-leap-1924/edit-timeline-spans'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13986565837

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13986565837
